### PR TITLE
Next implement erc165

### LINKED ISSUE #120: `TemplateLauncher.addTemplate` should check template contract signature/interface
[Full explanation](https://github.com/cryptonative-ch/aqua-subgraph/issues/92)

### PR DESCRIPTION
**Pull request by denciu**
### Summary 

Implements [ERC165](https://eips.ethereum.org/EIPS/eip-165) to prevent adding incompatible contracts to Template Laucher

## Description

Same motivation as per [Adam's PR](https://github.com/cryptonative-ch/mesa-smartcontracts/pull/121) but utilizing standardized solution suggested by Augusto. 
- `ITemplate` and `AquaTemplate` has been merged into abstract contract `AquaTemplate`,
- `AquaTemplate` implements ERC165 and adds `AquaTemplate.init.selector ^ AquaTemplate.templateName.selector` interface signature to supportedInterfaces list in constructor call
- Above signature is stored in `AquaTemplateId` contract, so it can be accessed from `TemplateLauncher`
- Fixes linting

This PR is concurrent to [Adam's PR](https://github.com/cryptonative-ch/mesa-smartcontracts/pull/121) therefore whichever PR is chosen, other should be closed.

Closes #120 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: contracts/templates/TemplateLauncher.sol [hunk 1]
```diff
@@ -120,6 +122,10 @@ contract TemplateLauncher is CloneFactory {
         isAllowedToAddTemplate
         returns (uint256)
     {
+        require(
+            ERC165Checker.supportsInterface(_template, _INTERFACE_ID_TEMPLATE),
+            "TemplateLauncher: TEMPLATE_INTERFACE_NOT_SUPPORTED"
+        );
```
Comment (adamazad) [inline contracts/templates/TemplateLauncher.sol:128]: Bullish 🐂

**Review by adamazad — APPROVED** [review]
Awesome work, @denciu!

File: contracts/templates/TemplateLauncher.sol [hunk 2]
```diff
@@ -40,7 +42,7 @@ contract TemplateLauncher is CloneFactory {
     address public participantListLaucher;
     bool public allowPublicTemplates;
 
-    modifier isTemplateManager {
+    modifier isTemplateManager() {
```
Comment (adamazad) [inline contracts/templates/TemplateLauncher.sol:45]: What's the difference between having and not having the `()`

Comment (denciu) [reply contracts/templates/TemplateLauncher.sol:45]: For some reason when I run linter it creates such quirks, I forgot to delete them, especially that it compiles normally.

Comment (adamazad) [reply contracts/templates/TemplateLauncher.sol:45]: Should we merge and update the audit's commit? @nicoelzer @jpkcambridge

**Comment by adamazad** [comment]
Should we merge and update the audit's ref? @nicoelzer @jpkcambridge

**Pull request merged**